File: core/src/main/scala/com/graphbrain/eco/nodes/StringNode.scala
package com.graphbrain.eco.nodes

import com.graphbrain.eco.{Contexts, Context, NodeType}

class StringNode(val value: String, lastTokenPos: Int= -1) extends ProgNode(lastTokenPos) {
  override def ntype(ctxt: Context) = NodeType.String
  override def stringValue(ctxts: Contexts, ctxt: Context) = value

  override def toString = "\"" + value + "\""

  override def equals(obj:Any) = obj match {
    case s: StringNode => s.value == value
    case _ => false
  }
}

File: core/src/main/java/com/graphbrain/eco/nodes/StringNode.java
package com.graphbrain.eco.nodes;

import com.graphbrain.eco.Context;
import com.graphbrain.eco.Contexts;
import com.graphbrain.eco.NodeType;

public class StringNode extends ProgNode {

    private String value;

    public StringNode(String value, int lastTokenPos) {
        super(lastTokenPos);
        this.value = value;
    }

    public StringNode(String value) {
        this(value, -1);
    }

    @Override
    public NodeType ntype(Context ctxt) {return NodeType.String;}

    @Override
    public void eval(Contexts ctxts) {
        for (Context c : ctxts.getCtxts())
            c.setRetString(this, value);
    }

    @Override
    public String toString() {
        return "\"" + value + "\"";
    }

    @Override
    public boolean equals(Object obj) {
        if (obj instanceof StringNode) {
            StringNode s = (StringNode)obj;
            return s.value.equals(value);
        }
        return false;
    }

    public String getValue() {
        return value;
    }
}